path_static_roles.go

```diff
@@ -247,17 +251,18 @@ func (b *backend) pathStaticRoleCreateUpdate(ctx context.Context, req *logical.R
 		// the queue
 
 		//TODO: Add retry logic
-		_, err = b.popFromRotationQueueByKey(name)
+		// We could be tracking a WAL ID for this role, ensure we keep it when
+		// it's re-added to the queue.
```

Comment:
Is this comment still applicable? When we're popping this item from the queue, it should already have the `walID` if that was previously set right?

Reply:
That's right. Maybe I've written it weirdly, but I wanted to leave a comment to highlight that it's important we keep the `item` returned from pop, unlike before.